extensions/dub-link-shortener/src/api/index.ts

```diff
@@ -0,0 +1,57 @@
+import axios, { AxiosRequestConfig } from "axios";
+import { BASE_API_URL } from "@utils/constants";
+import { apiKey, commandName, extensionName } from "@utils/env";
+import { DeleteLinkResponseBody, LinkSchema, TagSchema, WorkspaceId, WorkspaceSchema } from "@/types";
+import { captureException } from "@raycast/api";
+
+const hasHttps = (url: string) => url.startsWith("https://");
+const headers = {
+  Authorization: "Bearer " + apiKey,
+  "Content-Type": "application/json",
+  "user-agent": `raycast/${extensionName}/${commandName}`,
+};
+
+export const callApi = async <T>(config: AxiosRequestConfig) => {
+  return await axios({ ...config, headers })
+    .then(({ data }) => data as T)
+    .catch(({ response }) => {
+      const err = new Error(response?.data?.error?.message ?? "Unknown error");
+      captureException(err);
+      throw err;
+    });
+};
+
+export const getAllWorkspaces = async () => {
+  return await callApi<WorkspaceSchema[]>({ method: "GET", url: `${BASE_API_URL}/workspaces` });
+};
+
+export const createShortLink = async ({
+  originalUrl,
+  key,
+  workspaceId,
+  domain,
+  tagIds,
+  comments,
+}: { originalUrl: string; key?: string; domain?: string; tagIds?: string[]; comments?: string } & WorkspaceId) => {
+  const url = hasHttps(originalUrl) ? originalUrl : `https://${originalUrl}`;
+  return await callApi<LinkSchema>({
+    method: "POST",
+    url: `${BASE_API_URL}/links?workspaceId=${workspaceId}`,
+    data: { domain, url, key, tagIds, comments },
+  });
+};
```

Comment:
makes sense using the useForm raycast helper, although tbh wondering if this refactor should be skipped and go straight to using the Dub typescript SDK: https://www.npmjs.com/package/dub

Reply:
I thought about it but adding an external dependency might bring some maintenance costs during upgrade when the alternative is just a straight forward REST call. If it gets too heavy-handed with new command additions, then we can definitely leverage SDK. What do you think?

Reply:
I can add a todo for this one as well

Reply:
true, an additional pro for the SDK is the built in type-safety meanwhile for the raw REST you had to make your own typedefs (that can also go stale with upgrades)

but let's keep it with REST for now and see if things get too complex that it'd warrant using the SDK

Reply:
for sure I added a todo b/c it looked like the SDK is still in beta: https://github.com/dubinc/dub-node/blob/765e170c45a361de3ae62e0d19571ceca4a3f0f4/README.md#maturity